doc/source/dev/api-dev/nan_policy.rst

```diff
@@ -0,0 +1,168 @@
+collection of 1-d arrays, with the ``axis`` parameter specifying the axis
+along which the reduction is to be applied.  If, say::
+
+    func([1, 3, 4])     -> 10.0
+    func([2, -3, 8, 2]) ->  4.2
+    func([7, 8])        ->  9.5
+    func([])            -> -inf
+
+then::
+
+    func([[  1, nan,   3,   4],
+          [  2,  -3,   8,   2],
+          [nan,   7, nan,   8],
+          [nan, nan, nan, nan]], nan_policy='omit', axis=-1)
+
+must give the result::
+
+    np.array([10.0, 4.2, 9.5, -inf])
+
+
+Edge cases
+----------
+A function that implements the ``nan_policy`` parameter should gracefully
+handle the case where *all* the values in the input array(s) are ``nan``.
+The basic principle described above still applies::
+
+    func([nan, nan, nan], nan_policy='omit')
+
+should behave the same as::
+
+    func([])
+
+In practice, when adding ``nan_policy`` to an existing function, it is
+not unusual to find that the function doesn't already handle this case
+in a well-defined manner, and some thought and design may have to be
+applied to ensure that it works.  The correct behavior (whether that be
+to return ``nan``, return some other value, raise an exception, or something
+else) will be determined case-by-case.
+
+
+Why doesn't ``nan_policy`` also apply to ``inf``?
+--------------------------------------------------
+Although we learn in grade school that "infinity is not a number", the
+floating point values ``nan`` and ``inf`` are qualitatively different.
+The values ``inf`` and ``-inf`` act much more like regular floating
+point values than ``nan``.
+
+* One can compare ``inf`` to other floating point values and it behaves
+  as expected, e.g. ``3 < inf`` is True.
+* For the most part, arithmetic works "as expected" with ``inf``,
+  e.g. ``inf + inf = inf``, ``-2*inf = -inf``, ``1/inf = 0``,
+  etc.
+* Many existing functions work "as expected" with ``inf``:
+  ``np.log(inf) = inf``, ``np.exp(-inf) = 0``,
+  ``np.array([1.0, -1.0, np.inf]).min() = -1.0``, etc.
+
+So while ``nan`` almost always means "something went wrong" or "something
+is missing", ``inf`` can in many cases be treated as a useful floating
+point value.
+
+It is also consistent with the NumPy ``nan`` functions to not ignore
+``inf``::
+
+    >>> np.nanmax([1, 2, 3, np.inf, np.nan])
+    inf
+    >>> np.nansum([1, 2, 3, np.inf, np.nan])
+    inf
+    >>> np.nanmean([8, -np.inf, 9, 1, np.nan])
+    -inf
+
+
+Subtleties of ``nan_policy='propagate'``
+----------------------------------------
+In many case, ``nan_policy='propagate'`` is trivial to implement: just
+execute the code in the function without checking for ``nan``, and let the
+chips fall where they may.  There are two potential problems with this:
+
+* It may result in warnings that, for the user, are unexpected. E.g. "Why
+  are you warning me about invalid values in double scalars when I have
+  already told you to propagate the ``nan`` values?"
```

Comment:
```suggestion
execute the code in the function without checking for ``nan``.
There are potential problems with this approach. 

For example:

```

Reply:
@WarrenWeckesser I left a suggestion here to cut this part of the section given we both seem to agree it's not very clear and we can always clarify and add later.